### PR TITLE
Verification and Validation

### PR DESCRIPTION
**Pull request by Rikkola**
Next steps. 
Fixes two bugs really. The summit example @manstis did picked these both up. 

This PR adds support for:
- BRL condition/column
- Action/column cell having a null value
- Action/column cell having no value
- Action/column cell having more than one values

### CODE REVIEW

**Comment by manstis** [comment]
@Rikkola Nice small PR for me to start my day with ;-)

**Comment by Rikkola** [comment]
@manstis Hehe, have to keep pushing these, but I think this is the last one with this big impact. 
Again, only one who needs this ATM is me and I have it locally, so no hurry.

**Comment by manstis** [comment]
@Rikkola 3 failing tests (not sure why they should fail.. unless related to something in your PR?) Could you please confirm everything passes locally for you (with the latest fetch from master)?

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by Rikkola** [comment]
Could be that my merges broke something when I cherry picked from one topic branch to another. Checking.

**Comment by Rikkola** [comment]
Ok. I see them. I broke something not related to V&V.

**Comment by Rikkola** [comment]
Same 3 also fail in master.
(Still doesn't mean I didn't break them)

**Comment by Rikkola** [comment]
Fix on the way.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/cache/RuleInspector.java [hunk 1]
```diff
@@ -56,6 +71,44 @@ public RuleInspector( final Rule rule,
 
             patternInspectorList.add( patternInspector );
         }
+
+        updateBRLActionInspectors( rule.getActions()
+                                       .where( Action.superType().is( ActionSuperType.BRL_ACTION ) )
+                                       .select().all() );
+        rule.getActions()
+            .where( Action.superType().is( ActionSuperType.BRL_ACTION ) )
+            .listen().all( new AllListener<Action>() {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/cache/RuleInspector.java]: FYI, this type of construct can be simplified with lamdas to read ```.listen().all( (all)->updateBRLActionInspectors(all) );``` or even (IIRC) ```.listen().all( (all)->this::updateBRLActionInspectors );``` (which IMO is slightly *less* readable!)

**Comment by Rikkola** [comment]
Works on Idea, fails in CLI. 
Could be this.
https://groups.google.com/forum/embed/#!topic/mockito/LCWH3ykLJtA

**Comment by Rikkola** [comment]
Or not. Something similar.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/index/FieldConditionsBuilder.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                                                        conditionCol52,
+                                                        row.get( columnIndex ) );
+            field.getConditions().add( condition );
+            rule.getConditions().add( condition );
+        }
+    }
+
+    private Field resolveField( final ConditionCol52 conditionCol52 ) {
+        return Utils.resolveField( pattern,
+                                                conditionCol52.getFieldType(),
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/index/FieldConditionsBuilder.java]: Minor nit-pick, re: formatting.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/index/RuleBuilder.java [hunk 3]
```diff
@@ -59,14 +62,25 @@ private void resolvePatterns() {
...
-                                   pattern ).buildConditions();
+            new FieldConditionsBuilder( index,
+                                        model,
+                                        rule,
+                                        row,
+                                        utils,
+                                        pattern ).buildConditions( pattern52.getChildColumns() );
+        }
+
+        for ( final CompositeColumn<? extends BaseColumn> column : model.getConditions() ) {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/index/RuleBuilder.java]: You could possibly wrap the iterator over ```getPatterns()``` and ```getConditions()``` into a single loop and use ```instanceof Pattern52``` and ```instanceof BRLConditionColumn``` inside the same loop.

**Comment by manstis** [comment]
@Rikkola a few trivial-ignorable comments (mainly FYI)... when green I'll +1 and merge.

**Comment by manstis** [comment]
@Rikkola let me check the failing tests from CLI here too (w/o your PR.. I can fix as appropriate)

**Comment by manstis** [comment]
@Rikkola probably this "Actually the issue is that the anonymous class is not public, it is compiled by the Sun compiler as a package visible class" as the test sub-classes a class with a package-protected method and then spies on the sub-class.

**Comment by Rikkola** [comment]
Ah, I checked the class, but not the method. Well spotted.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/cache/RuleInspector.java (hunk 1 above)
Comment (Rikkola) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/cache/RuleInspector.java]: Tried that but GWT compile broke. Have you been able to do that on client side?

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/index/RuleBuilder.java (hunk 3 above)
Comment (Rikkola) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/index/RuleBuilder.java]: Ok. Good to know. I'll change this.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/cache/RuleInspector.java (hunk 1 above)
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/cache/RuleInspector.java]: @Rikkola Yes, I do use it quite a lot e.g. https://github.com/droolsjbpm/drools-wb/blob/master/drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/GuidedDecisionTablePresenter.java#L962

Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/cache/RuleInspector.java]: @Rikkola I find lamdas work OK (on the whole, although IIRC @porcelli had an issue with one use-case).. none of the Java 8 "stream" API is implemented in GWT2.8.0.beta1 so any use of that client-side is broken (but it's available in newer GWT2.8's)

Comment (Rikkola) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/cache/RuleInspector.java]: Ok, thank you. Need to check what I did wrong.

**Comment by Rikkola** [comment]
Fixed formatting and the iterator. I still need to rebase once #208 is in.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@Rikkola #208 is merged.

**Comment by manstis** [comment]
+1 to merge.

**Pull request merged**